File: api-backoffice/report/php/download.php
<?php

// Report - Download

$requestURL = "https://int-ecommerce.nexi.it/"; // URL
$requestURI = "ecomm/api/bo/downloadReport"; // URI

// Parametri per calcolo MAC
$apiKey = "<ALIAS>"; // Alias fornito da Nexi
$chiaveSegreta = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Chiave segreta fornita da Nexi
$idReport = "1"; // Id del report recuperato con l'api ecomm/api/bo/elencoReport
$timeStamp = (time()) * 1000;

// Calcolo MAC
$mac = sha1('apiKey=' . $apiKey . "timeStamp=" . $timeStamp . "idReport=" . $idReport . $chiaveSegreta);

// Parametri
$parametri = array(
    'apiKey' => $apiKey,
    'idReport' => $idReport,
    'timeStamp' => $timeStamp,
    'mac' => $mac
);

// Url da utilizzare per fare il download del report
$urlDownload = $requestURL . $requestURI . '?' . http_build_query($parametri);

File: api-backoffice/report/dotnet/download.cs
// Report - Download

using System;
using System.Collections.Generic;
using System.Text;
using System.Security.Cryptography;

namespace XPay
{
    class SampleXPayRequest
    {
        static void Main(string[] args)
        {

            // Alias e chiave segreta
            string ALIAS = "<ALIAS>"; // Sostituire con il valore fornito da Nexi
            string CHIAVESEGRETA = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

            string requestUrl = "https://int-ecommerce.nexi.it/ecomm/api/bo/downloadReport";

            string timeStamp = (Math.Round((DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalMilliseconds)).ToString();
            string idReport = "123";       

            // Calcolo MAC
            SampleXPayRequest sampleXPay = new SampleXPayRequest();
            string mac = sampleXPay.HashMac("apiKey=" + ALIAS + "timeStamp=" + timeStamp + "idReport=" + idReport + CHIAVESEGRETA);

            var requestParams = new Dictionary<string, string>();
            // Parametri obbligatori
            requestParams["apiKey"] = ALIAS;
            requestParams["timeStamp"] = importo;
            requestParams["idReport"] = divisa;
            requestParams["mac"] = mac;

            string uri =  "?";
            foreach (KeyValuePair<string, string> param in requestParams)
            {
                uri += param.Key + "=" + param.Value + "&";
            }

            string result = requestUrl + uri;
        }

        public string HashMac(string s)
        {
            byte[] bytes = Encoding.UTF8.GetBytes(s);

            var sha1 = SHA1.Create();
            byte[] hashBytes = sha1.ComputeHash(bytes);

            return HexStringFromBytes(hashBytes);
        }

        public static string HexStringFromBytes(byte[] bytes)
        {
            var sb = new StringBuilder();
            foreach (byte b in bytes)
            {
                var hex = b.ToString("x2");
                sb.Append(hex);
            }
            return sb.ToString();
        }

    }
}